src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts

```diff
@@ -0,0 +1,67 @@
+import { LoginService } from 'dotcms-js/dotcms-js';
+import { DotRenderedPageState, DotPageState } from './../../../shared/models/dot-rendered-page-state.model';
+import { DotRenderHTMLService } from './../../../../../api/services/dot-render-html/dot-render-html.service';
+import { DotRenderedPage } from '../../../shared/models/dot-rendered-page.model';
+import { Observable } from 'rxjs/Observable';
+import { Injectable } from '@angular/core';
+import { DotContentletLockerService } from '../../../../../api/services/dot-contentlet-locker/dot-contentlet-locker.service';
+import { PageMode } from '../../../shared/models/page-mode.enum';
+
+@Injectable()
+export class DotPageStateService {
+    constructor(
+        private dotRenderHTMLService: DotRenderHTMLService,
+        private dotContentletLockerService: DotContentletLockerService,
+        private loginService: LoginService
+    ) {}
+
+    /**
+     * Set the page state
+     *
+     * @param {DotRenderedPage} page
+     * @param {DotEditPageState} state
+     * @returns {Observable<any>}
+     * @memberof DotRenderHTMLService
+     */
+    set(page: DotRenderedPage, state: DotPageState): Observable<DotRenderedPageState> {
```

Comment:
Here we set a state (because the toolbar say so) and we return a whole new state to update everything.